internal/component/loki/source/file/tailer.go

```diff
@@ -47,12 +48,10 @@ type tailer struct {
 
 	componentStopping func() bool
 
-	mut      sync.RWMutex
-	stopping bool
-	tail     *tail.Tail
-	posquit  chan struct{} // used by the readLine method to tell the updatePosition method to stop
-	posdone  chan struct{} // used by the updatePosition method to notify when it stopped
-	done     chan struct{} // used by the readLine method to notify when it stopped
+	tail    *tail.Tail
+	posquit chan struct{} // used by the readLine method to tell the updatePosition method to stop
+	posdone chan struct{} // used by the updatePosition method to notify when it stopped
+	done    chan struct{} // used by the readLine method to notify when it stopped
```

Comment:
I wonder if all these channels have to be struct fields and whether all the goroutines have to be started in Run... it is harder to see what depends on what this way. WDYT about e.g. `readLines` starting the `updatePosition` goroutine and closing it before it closes?

Reply:
I was experimenting a bit with this before, not having them as struct fields but we have to pass them around.

I made it so that both readers works the same.

Run starts both `updatePosition` and `readlines`. When it stops either by it self or by context being canceled we stop `readlines` and wait for it. That one will close `posquit` that in turn. 

I can try to think if there is a better way to structure this

Reply:
[8e5e4b4](https://github.com/grafana/alloy/pull/3475/commits/8e5e4b415bc1826c5039e631c2ee507c16506d44) is an attempt to do this, wdyt?